sync/class.jetpack-sync-queue.php

```diff
@@ -325,18 +325,55 @@ function unlock() {
 	}
 
 	private function get_checkout_id() {
-		return get_transient( $this->get_checkout_transient_name() );
+		global $wpdb;
+		$checkout_value = $wpdb->get_var( 
+			$wpdb->prepare(
+				"SELECT option_value FROM $wpdb->options WHERE option_name = %s", 
+				$this->get_lock_option_name()
+			)
+		);
+
+		if ( $checkout_value ) {
+			list( $checkout_id, $timestamp ) = explode( ':', $checkout_value );
+			if ( $timestamp > time() ) {
+				return $checkout_id;
+			}
+		}
+
+		return false;
 	}
 
 	private function set_checkout_id( $checkout_id ) {
-		return set_transient( $this->get_checkout_transient_name(), $checkout_id, 5 * 60 ); // 5 minute timeout
+		global $wpdb;
+
+		$expires = time() + ( 5 * 60 );
```

Comment:
`( 5 * 60 )` could be a constant in Jetpack_Sync_Defaults, there's also a global `MINUTE_IN_SECONDS` constant in WP that we could use for this